rpc.go

```diff
@@ -111,7 +142,60 @@ func (r *rpcClient) close() error {
 	if r.cancelAuthRefresh != nil {
 		<-r.cancelAuthRefresh()
 	}
-	return r.client.Close()
+
+	// we don't want to interrupt the cleanup of these links.
+	r.resetLinkCache(context.Background())
```

Comment:
Do callers of `close()` have a context?  If so we should flow it through to here instead of using background.

Reply:
Unfortunately, no. 

The 'public' function is 'Close()', no context. I promise we won't do that in track 2. :)